service/controller/resource/cloudconfigblob/create.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	machinePool, err := r.getOwnerMachinePool(ctx, azureMachinePool.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if machinePool == nil {
+		return microerror.Mask(ownerReferenceNotSet)
+	}
+
+	credentialSecret, err := r.getCredentialSecret(ctx, &azureMachinePool)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	blobName := key.BlobName(&azureMachinePool, key.PrefixWorker())
+
+	var payload string
+	{
+		payload, err = r.getCloudConfigFromBootstrapSecret(ctx, machinePool)
+		if errors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "bootstrap CR or cloudconfig secret were not found")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if IsBootstrapCRNotReady(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "bootstrap CR is not ready")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var containerURL azblob.ContainerURL
+	{
+		containerURL, err = r.getContainerURL(ctx, credentialSecret, key.ClusterID(&azureMachinePool), key.StorageAccountName(&azureMachinePool))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring container object %#q contains bootstrap config", blobName))
+
+		_, err = blobclient.PutBlockBlob(ctx, blobName, payload, &containerURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured container object %#q contains bootstrap config", blobName))
+	}
+
+	return nil
+}
+
+// getCloudConfigFromBootstrapSecret returns the Bootstrap cloud config from the Bootstrap secret.
+func (r *Resource) getCloudConfigFromBootstrapSecret(ctx context.Context, machinePool *expcapiv1alpha3.MachinePool) (string, error) {
+	var err error
+	var bootstrapSecretName string
+	{
+		bootstrapSecretName, err = r.getBootstrapSecretName(ctx, machinePool)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	var cloudconfigSecret corev1.Secret
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Trying to find Secret containing bootstrap config %#q", bootstrapSecretName))
+		err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: machinePool.Namespace, Name: bootstrapSecretName}, &cloudconfigSecret)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	return string(cloudconfigSecret.Data[key.CloudConfigSecretKey]), nil
+}
+
+// getBootstrapSecretName will try to find Ignition CRs instead of Spark CRs when Ignition Operator is deployed.
+// It tries to find a Bootstrap CR which is named after the MachinePool. We may want to change it so we use `MachinePool.Spec.Template.Spec.Bootstrap`.
```

Comment:
[There are some fields in `MachinePool.Spec.Template.Spec`](https://github.com/kubernetes-sigs/cluster-api/blob/e3a4ed7977b6789e03cb9e4d6e5ca201e43a726e/api/v1alpha3/machine_types.go#L188-L209) that may be useful. Should we use them? I'm currently looking for a Spark CR that has the same name as the node pool, maybe I should use `ConfigRef` instead?